tsl/src/chunk_copy.c

```diff
@@ -420,16 +426,120 @@ chunk_copy_stage_create_empty_chunk_cleanup(ChunkCopy *cc)
 									  cc->dst_server->serverid);
 }
 
+static void
+chunk_copy_get_source_compressed_chunk_name(ChunkCopy *cc)
+{
+	char *cmd;
+	DistCmdResult *dist_res;
+	PGresult *res;
+
+	/* Get compressed chunk name on the source data node */
+	cmd = psprintf("SELECT c2.table_name "
+				   "FROM _timescaledb_catalog.chunk c1 "
+				   "JOIN _timescaledb_catalog.chunk c2 ON (c1.compressed_chunk_id = c2.id) "
+				   "WHERE c1.schema_name = %s and c1.table_name = %s",
+				   quote_literal_cstr(NameStr(cc->chunk->fd.schema_name)),
+				   quote_literal_cstr(NameStr(cc->chunk->fd.table_name)));
+	dist_res =
+		ts_dist_cmd_invoke_on_data_nodes(cmd, list_make1(NameStr(cc->fd.source_node_name)), true);
+	res = ts_dist_cmd_get_result_by_node_name(dist_res, NameStr(cc->fd.source_node_name));
+	if (PQresultStatus(res) != PGRES_TUPLES_OK)
+		ereport(ERROR,
+				(errcode(ERRCODE_CONNECTION_EXCEPTION), errmsg("%s", PQresultErrorMessage(res))));
+
+	/* Set compressed chunk name of the source data node */
+	if (PQntuples(res) != 0)
+		snprintf(cc->fd.compressed_chunk_name.data,
```

Comment:
Can assert that we got ONE result from the other side.

Reply:
Not sure that Assert is a good way to deal here, but I think we should generate an error at least